pubsub/gcp/pubsub/pubsub.go

```diff
@@ -76,9 +82,37 @@ type WhatNow struct {
 	Type string `json:"type"`
 }
 
+var topicCacheRefreshInterval = 5 * time.Hour
+
 // NewGCPPubSub returns a new GCPPubSub instance.
 func NewGCPPubSub(logger logger.Logger) pubsub.PubSub {
-	return &GCPPubSub{logger: logger, closeCh: make(chan struct{})}
+	client := &GCPPubSub{
+		logger:     logger,
+		closeCh:    make(chan struct{}),
+		topicCache: make(map[string]cacheEntry),
+		lock:       &sync.RWMutex{},
+	}
+	return client
+}
+
+func (g *GCPPubSub) periodicCacheRefresh() {
+	ticker := time.NewTicker(topicCacheRefreshInterval)
```

Comment:
Should probably be something like half of the time or less, because otherwise values can be cached for a lot longer. For example, assume a value is created 1s after the ticker, then it will live for almost 10 hours (minus 1s)

Reply:
```suggestion
	ticker := time.NewTicker(topicCacheRefreshInterval / 5)
```